src/main/java/frc/robot/commands/ChargeStationBalance.java

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.subsystems.DriveTrain;
+import edu.wpi.first.math.geometry.Rotation2d;
+
+public class ChargeStationBalance extends CommandBase {
+
+  //private final Rotation2d m_rampAngle = Rotation2d.fromDegrees(-10); //useful if we are going to automate going on the ramp
+  private final Rotation2d m_balancedError = Rotation2d.fromDegrees(2.5); //error for what counts as balanced
+  private final double m_balancedDegrees = 0;
+  private final double m_balanceKP = 0.02; //change to control how fast robot drives during balancing
+  private final double m_maxMPS = 0.75;
+  private final double m_balanceSeconds = 1; //how many seconds the robot has to be balanced before stopping
+
+  private DriveTrain m_driveTrain;
+  private enum m_stages {
+    GROUND,
+    CHARGESTATION
+  }
+  private m_stages m_stage = m_stages.GROUND;
+  private Boolean m_balanced = false;
+  private final Timer m_timer = new Timer();
+  private double m_curTime = 0;
+
+  /** Creates a new ChargeStationBalance. */
+  public ChargeStationBalance() {
+    // Use addRequirements() here to declare subsystem dependencies.
+    addRequirements(m_driveTrain);
+  }
+
+  // Called when the command is initially scheduled.
+  @Override
+  public void initialize() {
+    m_timer.reset();
+    m_timer.start();
+  }
```

Comment:
You should initialize m_stage here. Always think about the possiblity of re-running the command.

Reply:
Other variables should be initialized here: m_curTime, m_balanced, etc. Every variable to you gave a value to in the constructor should just be initialized here.